### PR TITLE
UXPROD-3903: Update schemas to support new bursar export feature

### PR DESCRIPTION
**Pull request by ncovercash**
[UXPROD-3903](https://issues.folio.org/browse/UXPROD-3903): This PR will merge team @folio-org/bama's changes in folio-export-common to support the new bursar export configuration feature.

> [!WARNING]
> This PR must be merged before https://github.com/folio-org/mod-data-export-spring/pull/270 and https://github.com/folio-org/mod-data-export-worker/pull/485.

_Note: this PR will be squashed on merge; please ignore the lonh commit history._ This PR has been merged before in #55, reverted in #59, put back in #61, and finally reverted again in #63. **It has received reviews as part of the previous PRs**.

### CODE REVIEW

**Review by danetsao — APPROVED** [review]

**Review by khandramai — APPROVED** [review]

File: schemas/bursar-export/bursarExportJob.json [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    },
+    "groupByPatronFilter": {
+      "description": "Filter to use for the export when grouping by patron",
+      "$ref": "filter/bursarExportFilterAggregate.json"
+    },
+    "header": {
+      "description": "Header format for the export file",
+      "type": "array",
+      "items": {
+        "$ref": "tokens/bursarExportHeaderFooter.json"
```
Comment (NikitaSedyx) [inline schemas/bursar-export/bursarExportJob.json:21]: as on UI, IMO it's a bad name when you place 2 things like HeaderFooter

File: schemas/bursar-export/bursarExportJob.json [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    },
+    "groupByPatron": {
+      "description": "Choose whether to group fee/fines by patron",
+      "type": "boolean"
+    },
+    "groupByPatronFilter": {
+      "description": "Filter to use for the export when grouping by patron",
+      "$ref": "filter/bursarExportFilterAggregate.json"
+    },
+    "header": {
```
Comment (NikitaSedyx) [inline schemas/bursar-export/bursarExportJob.json:17]: it's not self-descriptive, our entity is Job (in this case BursarExport), what is the header in context of job

why not to call it exportHeaderFormat? and in description to add more info what formats, tokens etc can be used there

as for me, current description just explains name

File: schemas/bursar-export/filter/bursarExportFilterAge.json [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    "numDays": {
+      "description": "Number of days",
+      "type": "integer",
+      "minimum": 1
+    },
+    "condition": {
+      "description": "Condition to get days range",
+      "type": "string",
+      "enum": [
+        "LESS_THAN_EQUAL",
```
Comment (NikitaSedyx) [inline schemas/bursar-export/filter/bursarExportFilterAge.json:19]: just a question, does it make sense to create common enum for conditions? I see 3 schemas use the same set

File: schemas/jobWithLegacyBursarParameters.json [hunk 4]
```diff
@@ -0,0 +1,101 @@
+{
```
Comment (NikitaSedyx) [inline schemas/jobWithLegacyBursarParameters.json:1]: why do we need this schema?

Comment (ncovercash) [reply schemas/jobWithLegacyBursarParameters.json:1]: So this one can be autogenerated, too. @danetsao please confirm we do need this for mdes.

File: schemas/bursar-export/bursarExportJob.json (hunk 1 above)
Comment (NikitaSedyx) [reply schemas/bursar-export/bursarExportJob.json:21]: correct me if I'm wrong
header and footer and actually headerTokens/Formats/Whatewer and it contains array of different kinds tokens (conditional, date, etc)

if it's true, mb it makes sense to make clear names?

-----

what I read from the current schema: there is a job with header/footer that contains array of smth HeaderFooter

File: schemas/jobWithLegacyBursarParameters.json (hunk 4 above)
Comment (danetsao) [reply schemas/jobWithLegacyBursarParameters.json:1]: Yes it is needed. Here is the [schema](https://github.com/folio-org/mod-data-export-spring/pull/270/files#diff-cbf72f961becc6965ae285c6984b419f2a60684eb6a96665f3ae18bd8273a10f) and [file](https://github.com/folio-org/mod-data-export-spring/pull/270/files#diff-cbf72f961becc6965ae285c6984b419f2a60684eb6a96665f3ae18bd8273a10f:~:text=src/main/java/org/folio/de/entity/bursarlegacy/JobWithLegacyBursarParameters.java), and it's usage in [BursarMigrationService](https://github.com/folio-org/mod-data-export-spring/pull/270/files#diff-f545f1994f6119b66c9d8fb6cd9dac7565c5cc26651240d16dadd58f742f73f5).

Comment (NikitaSedyx) [reply schemas/jobWithLegacyBursarParameters.json:1]: I'm not sure you need this migration, let's keep this thread open before we understand that this migration is correct and required

File: schemas/bursar-export/bursarExportJob.json (hunk 2 above)
Comment (ncovercash) [reply schemas/bursar-export/bursarExportJob.json:17]: What would you propose instead? I'm not sure what name would be any clearer...

Comment (NikitaSedyx) [reply schemas/bursar-export/bursarExportJob.json:17]: I already did a proposal in the first comment + in another [thread](https://github.com/folio-org/folio-export-common/pull/64#discussion_r1478325553), just waiting a confirmation that my understanding is correct

File: schemas/bursar-export/bursarExportJob.json (hunk 1 above)
Comment (ncovercash) [reply schemas/bursar-export/bursarExportJob.json:21]: IMO the `tokens/` directory is sufficient

Comment (NikitaSedyx) [reply schemas/bursar-export/bursarExportJob.json:21]: do we read directories working with API? @ncovercash naming should be improved, it affects both FE and BE

Comment (NikitaSedyx) [reply schemas/bursar-export/bursarExportJob.json:21]: what's the problem to call headerTokens/exportHeaderTokens? and across repos code will be more readable and clear

and bursarExportHeaderFooter.json can be just bursarTokens.json, not smth headerFooter

**Pull request merged**